ibis/backends/dask/execution/strings.py

```diff
@@ -0,0 +1,318 @@
+    return execute_substring_series_series(
+        op, data, start, dd.from_array(np.repeat(length, len(start))), **kwargs
+    )
+
+
+@execute_node.register(ops.Substring, dd.Series, integer_types, dd.Series)
+def execute_string_substring_int_series(op, data, start, length, **kwargs):
+    return execute_substring_series_series(
+        op,
+        data,
+        dd.from_array(np.repeat(start, len(length))),
+        length,
+        **kwargs,
+    )
+
+
+# TODO - substring - #2553
+@execute_node.register(ops.Substring, dd.Series, dd.Series, dd.Series)
+def execute_substring_series_series(op, data, start, length, **kwargs):
+    end = start + length
+
+    # TODO - this is broken
+    def iterate(
+        value, start_iter=start.iteritems(), end_iter=end.iteritems(),
+    ):
+        _, begin = next(start_iter)
+        _, end = next(end_iter)
+        if (begin is not None and isnull(begin)) or (
+            end is not None and isnull(end)
+        ):
+            return None
+        return value[begin:end]
+
+    return data.map(iterate)
+
+
+# TODO - grouping - #2553
+@execute_node.register(ops.StringSQLLike, ddgb.SeriesGroupBy, str, str)
+def execute_string_like_series_groupby_string(
+    op, data, pattern, escape, **kwargs
+):
+    return execute_string_like_series_string(
+        op, data.obj, pattern, escape, **kwargs
+    ).groupby(data.grouper.groupings)
+
+
+# TODO - aggregations - #2553
+@execute_node.register(
+    ops.GroupConcat, dd.Series, str, (dd.Series, type(None))
+)
+def execute_group_concat_series_mask(
+    op, data, sep, mask, aggcontext=None, **kwargs
+):
+    return aggcontext.agg(
+        data[mask] if mask is not None else data,
+        lambda series, sep=sep: sep.join(series.values),
+    )
+
+
+# TODO - grouping - #2553
+@execute_node.register(ops.GroupConcat, ddgb.SeriesGroupBy, str, type(None))
+def execute_group_concat_series_gb(
+    op, data, sep, _, aggcontext=None, **kwargs
+):
+    custom_group_concat = dd.Aggregation(
+        name='custom_group_concat',
+        chunk=lambda s: s.apply(list),
+        agg=lambda s0: s0.apply(
+            lambda chunks: sep.join(
+                str(s) for s in itertools.chain.from_iterable(chunks)
+            )
+        ),
+    )
+    return data.agg(custom_group_concat)
+
+
+# TODO - grouping - #2553
+@execute_node.register(
+    ops.GroupConcat, ddgb.SeriesGroupBy, str, ddgb.SeriesGroupBy
+)
+def execute_group_concat_series_gb_mask(
+    op, data, sep, mask, aggcontext=None, **kwargs
+):
+    def method(series, sep=sep):
+        return sep.join(series.values.astype(str))
+
+    return aggcontext.agg(
+        data,
+        lambda data, mask=mask.obj, method=method: method(
+            data[mask[data.index]]
+        ),
+    )
+
+
+@execute_node.register(ops.StringAscii, dd.Series)
+def execute_string_ascii(op, data, **kwargs):
+    output_meta = pandas.Series([], dtype=np.dtype('int32'), name=data.name)
```

Comment:
Hmm.. what should the type of `meta` be in general? looks like sometimes it's a pd.Series sometimes it's a dict?

Reply:
https://docs.dask.org/en/latest/dataframe-design.html#metadata. A few different ways to apply meta are all equally valid. My reads is you can:

1. Provide an empty `Pandas object with appropriate dtypes and names`. 
2. A descriptive meta, which can be: 
```
A dict of {name: dtype} or an iterable of (name, dtype) specifies a DataFrame. Note that order is important: the order of the names in meta should match the order of the columns
A tuple of (name, dtype) specifies a series
A dtype object or string (e.g. 'f8') specifies a scalar
```